script/entrypoint

```diff
@@ -32,6 +32,5 @@ elif [ "$1" = "query" ]; then
 else
     # KubernetesPodOperator will extract the contents of /airflow/xcom/return.json as an xcom
     # if the xcom_push parameter is true
-    mkdir -p /airflow/xcom && touch /airflow/xcom/return.json
     exec "$@" > /airflow/xcom/return.json
```

Comment:
> I'd also like to not lose error logs in the exec case i.e. use `&>`

Good point, I think we should be preserving stdout unless specifically requested. 

I think we should handle this before this conditional block via `exec`, e.g. add `if "${XCOM_PUSH:-false}"; then exec > /airflow/xcom/return.json; fi` around line 19, and remove the redirect here.
```suggestion
     exec "$@"
```

Reply:
Any reason we'd prefer that over using `tee`?

Reply:
I don't know much about the xcom stuff but as long as stdout/err are preserved on normal jenkins runs, I'm easy as to the underlying mechanism. To that end `a+w` (assuming umask is fine) and `tee` would suffice.

Reply:
> Any reason we'd prefer that over using `tee`?

it applies to the whole script, rather than just this line.

Reply:
but i'd settle for `tee`